src/LazyRendering/View/MainLayout.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\LazyRendering\View;
+
+use App\Asset\AppAsset;
+use App\Widget\PerformanceMetrics;
+use Generator;
+use Psr\Http\Message\RequestInterface;
+use Yiisoft\Assets\AssetManager;
+use Yiisoft\Html\Html;
+
+class MainLayout
+{
+    private AssetManager $assetManager;
+    private ?string $title = null;
+    private array $metaTags = [
+        ['charset' => 'UTF-8'],
+        [
+            'name'    => 'viewport',
+            'content' => 'width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0',
+        ],
+        ['http-equiv' => 'X-UA-Compatible', 'content' => 'ie=edge'],
+    ];
+
+    public function __construct(AssetManager $assetManager)
+    {
+        $this->assetManager = $assetManager;
+    }
+
+    public function render(iterable $content, RequestInterface $request): iterable
+    {
+        ob_start();
+        foreach ($this->renderPage($content) as $contentPart) {
+            yield  ob_get_clean() . $contentPart;
+            ob_start();
+        }
+        ob_end_clean();
+    }
+
+    protected function renderPage(iterable $content): Generator
+    {
+        $this->assetManager->register([AppAsset::class]);
+
+        echo '<html><head>' . ($this->title !== null ? '<title>' . Html::encode($this->title) . '</title>' : '');
```

Comment:
Is mix of `echo` and `yield` intended?

Reply:
Yes
Each 'yield' means the end of a chunk in chucked output. I use 'echo' to combine several small chunks that don't use i/o into one